bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/DataPointByteValue.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ism8.server;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DataPointByteValue} is the data points for byte values
+ *
+ * @author Hans-Reiner Hoffmann - Initial contribution
+ */
+public class DataPointByteValue extends DataPointBase<Byte> {
+    private final Logger logger = LoggerFactory.getLogger(DataPointByteValue.class);
+
+    public DataPointByteValue(int id, String knxDataType, String description) {
+        super(id, knxDataType, description);
+    }
+
+    @Override
+    public String getValueText() {
+        return this.getValue().toString();
+    }
+
+    @Override
+    public void processData(byte[] data) {
+        if (this.checkProcessData(data)) {
+            if (data[3] != 1 && data.length < 4) {
```

Comment:
The length <4 must be <=4. The query is not to prevent the access to [3] in the same line, but the access in line 43 (data[4]). I placed on purpose this sequence to not be misleaded by the length check. The minimum length of 4 is checked in the IF-statement of line 38. The same mistake I had made in the other data points. I have rechecked and corrected all of them.